scipy/sparse/tests/test_base.py

```diff
@@ -3610,22 +3612,22 @@ def test_add_sub(self):
                 assert_array_equal(A - Bsp,D1)          # check dense - sparse
 
     def test_mu(self):
-        self.__arith_init()
+        __A, __B, __Asp, __Bsp = self.__arith_init()
 
         # basic tests
-        assert_array_equal((self.__Asp @ self.__Bsp.T).toarray(),
-                           self.__A @ self.__B.T)
+        assert_array_equal((__Asp @ __Bsp.T).toarray(),
+                            __A @ __B.T)
 
         for x in supported_dtypes:
             with np.errstate(invalid="ignore"):
-                A = self.__A.astype(x)
+                A = __A.astype(x)
             Asp = self.spcreator(A)
             for y in supported_dtypes:
                 if np.issubdtype(y, np.complexfloating):
-                    B = self.__B.astype(y)
+                    B = __B.astype(y)
                 else:
                     with np.errstate(invalid="ignore"):
-                        B = self.__B.real.astype(y)
+                        B = __B.real.astype(y)
```

Comment:
do the `sparse` folks have any concerns about changes in "meaning" here and above? I guess this is related to `class` objects having various tricky properties in parallel situations

Reply:
This looks fine to me. Yes, just avoid multiple threads trying to set the same attribute on the class. There's not much point really to all the `self.`'s, that just looks wrong.